clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/MigrateServerGroupStrategy.java

```diff
@@ -0,0 +1,271 @@
+        .getAWSServerGroupNameResolver()
+        .resolveNextServerGroupName(names.getApp(), names.getStack(), names.getDetail(), false);
+
+      result.setServerGroupNames(Collections.singletonList(targetName));
+    }
+    migrateResult.setServerGroupName(result.getServerGroupNames().get(0));
+    migrateResult.setLoadBalancers(targetLoadBalancers);
+    migrateResult.setSecurityGroups(targetSecurityGroups);
+    return migrateResult;
+  }
+
+  private BasicAmazonDeployDescription generateDescription(ServerGroupLocation source,
+                                                           BasicAmazonDeployDescription deployDescription) {
+    return getBasicAmazonDeployHandler().copySourceAttributes(
+      getRegionScopedProviderFactory().forRegion(source.getCredentials(), source.getRegion()), source.getName(),
+      false, deployDescription);
+  }
+
+  private static Source getSource(ServerGroupLocation source) {
+    Source deploySource = new Source();
+    deploySource.setAccount(source.getCredentialAccount());
+    deploySource.setRegion(source.getRegion());
+    deploySource.setAsgName(source.getName());
+    return deploySource;
+  }
+
+  private static Capacity getCapacity() {
+    Capacity capacity = new Capacity();
+    capacity.setMin(0);
+    capacity.setMax(0);
+    capacity.setDesired(0);
+    return capacity;
+  }
+
+  protected List<MigrateSecurityGroupResult> generateTargetSecurityGroups(LaunchConfiguration sourceLaunchConfig,
+                                                                          ServerGroupLocation source,
+                                                                          ServerGroupLocation target,
+                                                                          MigrateServerGroupResult result,
+                                                                          boolean dryRun) {
+
+    sourceLaunchConfig.getSecurityGroups().stream()
+      .filter(g -> !sourceLookup.getSecurityGroupById(source.getCredentialAccount(), g, source.getVpcId()).isPresent())
+      .forEach(m -> result.getWarnings().add("Skipping creation of security group: " + m
+        + " (could not be found in source location)"));
+
+    List<String> securityGroupNames = sourceLaunchConfig.getSecurityGroups().stream()
+      .filter(g -> sourceLookup.getSecurityGroupById(source.getCredentialAccount(), g, source.getVpcId()).isPresent())
+      .map(g -> sourceLookup.getSecurityGroupById(source.getCredentialAccount(), g, source.getVpcId()).get())
+      .map(g -> g.getSecurityGroup().getGroupName())
+      .collect(Collectors.toList());
+
+    List<MigrateSecurityGroupResult> targetSecurityGroups = securityGroupNames.stream().map(group ->
+      getMigrateSecurityGroupResult(source, target, sourceLookup, targetLookup, dryRun, group)
+    ).collect(Collectors.toList());
+
+    if (getDeployDefaults().getAddAppGroupToServerGroup()) {
```

Comment:
creation of this group will happen automatically during the deploy

I guess the case this would handle if the source group exists but hasn't already been attached to the source asg to ensure that the source rules ended up on the target when it was created. Not sure if that is a case we would encounter or not though